notification.go

```diff
@@ -0,0 +1,80 @@
+package watchdog
+
+import "sync"
+
+var (
+	gcNotifeeMutex sync.Mutex
+	gcNotifees     []notifeeEntry
+
+	forcedGCNotifeeMutex sync.Mutex
+	forcedGCNotifees     []notifeeEntry
+)
+
+// RegisterPostGCNotifee registers a function that is called every time a GC has happened,
+// both GC runs triggered by the Go runtime and by watchdog.
+// The unregister function returned can be used to unregister this notifee.
+func RegisterPostGCNotifee(f func()) (unregister func()) {
+	gcNotifeeMutex.Lock()
+	defer gcNotifeeMutex.Unlock()
+
+	var id int
+	if len(gcNotifees) > 0 {
+		id = gcNotifees[len(gcNotifees)-1].id + 1
```

Comment:
After deletion the is will duplicate leading to weirdness around unregistration.

Reply:
Can you elaborate? The `id` is only used internally, and the logic have here guarantees that at any given moment each `id` is only used once.
As long as the caller doesn't call `unregister` multiple times, this will work fine, won't it?